diff -ur init-nitro-rc-orig/script/functions.in init-nitro-rc/script/functions.in
--- init-nitro-rc-orig/script/functions.in	2025-11-10 20:33:00.000000000 +0000
+++ init-nitro-rc/script/functions.in	2026-01-30 10:11:01.656760324 +0000
@@ -16,7 +16,7 @@
         USECOLOR=""
     elif [[ -t 0 ]]; then
         # stty will fail when stdin isn't a terminal
-        STAT_COL=$(stty size)
+        STAT_COL=$(stty size 2>/dev/null || echo 80)
         # stty gives "rows cols"; strip the rows number, we just want columns
         STAT_COL=${STAT_COL##* }
     elif tput cols &>/dev/null; then
@@ -168,7 +168,7 @@
 
 print_leave(){
     # avoid staircase effect
-    stty onlcr
+    stty onlcr 2>/dev/null || true
     echo " "
     printhl "Initiating shutdown\n"
     echo " "
--- init-nitro-rc-orig/sv.d/remount-root.in	2026-01-30 15:12:04.835213435 +0000
+++ init-nitro-rc/sv.d/remount-root.in	2026-01-30 15:17:47.746425027 +0000
@@ -11,8 +11,12 @@
         stat_done remount-root
         ;;
     stop)
-        stat_busy "Remounting root filesystem ro"
-        mount -o remount,ro / || stat_die remount-root
+        if findmnt -n -o OPTIONS / | grep -qE "^ro,|,ro,|,ro$"; then
+            stat_busy "Filesystem / already ro"
+        else
+            stat_busy "Remounting root filesystem ro"
+            mount -o remount,ro / || stat_die remount-root
+        fi
         sync
         rm_daemon remount-root
         stat_done remount-root
